webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import URI from 'urijs';
+import { Spinner } from 'patternfly-react';
+import Table from '../common/table';
+import { STATUS } from '../../constants';
+import './ModelsTable.css';
+
+class ModelsTable extends React.Component {
+  componentDidMount() {
+    this.props.updateSearch(this.props.data);
+  }
+  render() {
+    const {
+      schema,
+      perPage,
+      page,
+      searchQuery,
+      results,
+      sortBy,
+      sortOrder,
+    } = this.props;
+    return this.props.status === STATUS.PENDING ? (
+      <div className="tableloader">
+        <Spinner size="lg" loading />
+      </div>
+    ) : (
+      <Table
+        columns={schema({
+          url: new URI('models.json')
+            .setSearch('per_page', perPage)
+            .setSearch('page', page)
+            .setSearch('search', searchQuery)
+            .toString(),
+          sortTable: this.props.sortTable,
+          sortBy,
+          sortOrder,
+        })}
+        rows={results}
+        search={!!searchQuery}
+        emptyState={{
+          header: __('Models'),
+          description: __('Hardware Models...'),
+          documentation: {},
+          action: {
+            url: '#',
+            title: '',
+          },
+        }}
+      />
+    );
+  }
+}
+
+ModelsTable.propTypes = {
```

Comment:
You are missing some props.

Reply:
done.

Reply:
You still missing the `status` prop.
Could you also add `defaultProps` for the not required props?